### PR TITLE
gh-112139: Add `inspect.Signature.format` and use it in `pydoc`

### PR DESCRIPTION
**Pull request by sobolevn**
What do you think about `__str__` with a parameter? Or should we add an alias for this, like `def to_string(self, *, pretty)`?

* Issue: gh-112139

### CODE REVIEW

File: Lib/inspect.py [hunk 1]
```diff
@@ -3352,7 +3352,11 @@ def __str__(self):
             # flag was not reset to 'False'
             result.append('/')
 
-        rendered = '({})'.format(', '.join(result))
+        params = ', '.join(result)
+        if pretty and len(params) > 78:  # 80 - '(' - ')'
```
Comment (JelleZijlstra) [inline Lib/inspect.py]: pydoc will also display the function name on the same line.

File: Lib/inspect.py [hunk 2]
```diff
@@ -3352,7 +3352,11 @@ def __str__(self):
...
+            rendered = '(\n    {}\n)'.format(',\n    '.join(result))
+        else:
+            rendered = '({})'.format(params)
```
Comment (JelleZijlstra) [inline Lib/inspect.py]: Why not an f-string?

File: Lib/test/test_pydoc.py [hunk 3]
```diff
@@ -870,6 +870,65 @@ class B(A)
         for expected_line in expected_lines:
             self.assertIn(expected_line, as_text)
 
+    def test_long_signatures(self):
+        from typing import Callable, Literal, Annotated
```
Comment (JelleZijlstra) [inline Lib/test/test_pydoc.py]:
```suggestion
        from collections.abc import Callable
        from typing import Literal, Annotated
```

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, it's great this is relatively simple to implement.

Should we make this a public, supported feature of `inspect.signature`? It may be useful for users other than `pydoc`. For example, we could add a `__format__`. Then perhaps something like `f"{sig:78p}"` could be used.

File: Lib/inspect.py (hunk 1 above)
Comment (sobolevn) [reply Lib/inspect.py]: Indeed!

**Comment by sobolevn** [comment]
Yes, I think we should add a new method called `format` to `inspect.Signature` and properly document it.

File: Doc/library/inspect.rst [hunk 4]
```diff
@@ -752,6 +752,17 @@ function.
...
       :func:`copy.replace`.
 
+   .. method:: format(*, max_width=None)
+
+      Convert signature object to string.
+
+      If *max_width* integer is passed,
+      signature will try to fit into the *max_width*.
+      If signature is longer than *max_width*,
+      all parameters will be on separate lines.
```
Comment (JelleZijlstra) [inline Doc/library/inspect.rst]:
```suggestion
      If *max_width* is passed, the method will attempt to fit
      the signature into lines of at most *max_width* characters.
      If the signature is longer than *max_width*,
      all parameters will be on separate lines.
```

File: Lib/test/test_inspect/test_inspect.py [hunk 5]
```diff
@@ -3818,19 +3828,59 @@ def test(a_po, /, *, b, **kwargs):
...
+
+    def test_signature_format(self):
+        from typing import Annotated, Literal
+
+        def func(x: Annotated[int, 'meta'], y: Literal['a', 'b'], z: 'LiteralString'):
+            pass
+
+        expected_singleline = "(x: Annotated[int, 'meta'], y: Literal['a', 'b'], z: 'LiteralString')"
+        expected_multiline = """(
+    x: Annotated[int, 'meta'],
```
Comment (JelleZijlstra) [inline Lib/test/test_inspect/test_inspect.py:3864]: Can you have a test that includes `/` or `*` and check that it's put on its own line?

**Review by JelleZijlstra — COMMENTED** [review]
Thank you! Sorry I forgot about reviewing this PR.

File: Lib/test/test_inspect/test_inspect.py (hunk 5 above)
Comment (sobolevn) [reply Lib/test/test_inspect/test_inspect.py:3864]: Done.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**